code/game/objects/items/weapons/storage/holster.dm

```diff
@@ -0,0 +1,269 @@
+
+/obj/item/clothing/accessory/holster/on_removed(mob/user as mob)
+	QDEL_NULL(holster)
+	..()
+
+//For the holster hotkey
+//This verb is universal to any subtype of pouch/holster.
+/obj/item/storage/pouch/holster/verb/holster_verb()
+	set name = "Holster"
+	set category = "Object"
+	set src = usr.contents
+	if(!ishuman(usr))
+		return
+	if(usr.stat)
+		return
+
+	var/mob/living/carbon/human/H = usr
+
+	//List of priorities for holster hotkey. Back, suit store, belt, left pocket, right pocket.
+	//Back is for future holsters, if anyone decides to add those.
+	var/list/holster_priority = list(
+	H.back,
+	H.s_store,
+	H.belt,
+	H.l_store,
+	H.r_store
+	)
+
+	for(var/i=1, i<holster_priority.len+1, ++i)
+		var/slot = holster_priority[i]
+
+		if(istype(slot, /obj/item/storage/pouch/holster))
+			var/obj/item/storage/pouch/holster/holster = slot
+
+			if(istype(usr.get_active_hand(),/obj))
+				if(holster.contents.len < holster.storage_slots)//putting items in holsters
+					holster.attackby(usr.get_active_hand(), usr)
+					break
+
+				//else
+				to_chat(usr, SPAN_NOTICE("All your holsters are occupied."))
+
+			else
+				if(holster.contents.len)//pulling items out of holsters
+					holster.attack_hand(usr)
+					break
+
+				//else
+				to_chat(usr, SPAN_NOTICE("You don't have any occupied holsters."))
+
+
+/obj/item/storage/pouch/holster/attack_hand()
+	if(contents.len)
+		var/obj/item/I = contents[contents.len]
+		if(istype(I))
+			if(usr.a_intent == I_HURT)
+				usr.visible_message(
+					SPAN_DANGER("[usr] draws \the [I], ready to fight!"),
+						SPAN_WARNING("You draw \the [I], ready to fight!")
+					)
+			else
+				usr.visible_message(
+					SPAN_NOTICE("[usr] draws \the [I], pointing it at the ground."),
+					SPAN_NOTICE("You draw \the [I], pointing it at the ground.")
+					)
+			add_fingerprint(usr)
+			playsound(usr, "[src.sound_out]", 75, 0)
+			update_icon()
+			w_class = initial(w_class)
+			name = "[initial(name)]"
+	..()
+
+/obj/item/storage/pouch/holster/attackby(obj/item/I, mob/user)
+	add_fingerprint(usr)
+	w_class = max(w_class, src.w_class)
+	name = "occupied [initial(name)]"
+	playsound(usr, "[src.sound_in]", 75, 0)
+	update_icon()
+	..()
+
+/obj/item/storage/pouch/holster/update_icon()
+	..()
+	cut_overlays()
+	if(contents.len)
+		overlays += image('icons/inventory/pockets/icon.dmi', "pistol_layer")
+
+/obj/item/storage/pouch/holster/baton/update_icon()
+	..()
+	cut_overlays()
+	if(contents.len)
+		overlays += image('icons/inventory/pockets/icon.dmi', "baton_layer")
+
+/obj/item/storage/pouch/holster/belt/update_icon()
+	..()
+	cut_overlays()
+	if(contents.len)
+		overlays += image('icons/inventory/pockets/icon.dmi', "revolver_layer[contents.len]")
+
+/obj/item/storage/pouch/holster/belt/sheath/update_icon()
+	var/icon_to_set
+	for(var/obj/item/SW in contents)
+		icon_to_set = SW.icon_state
+	icon_state = "sheath_[contents.len ? icon_to_set :"0"]"
+	item_state = "sheath_[contents.len ? icon_to_set :"0"]"
+	..()
```

Comment:
Wouldn't that add revolver overlay to sheaths?

Reply:
Sheath inherits from belt holster, so that `..()` would call `/obj/item/storage/pouch/holster/belt/update_icon()`.

Reply:
it works properly on local, so I don't understand what you are talking about. icons_to_set is not declared in /obj/item/storage/pouch/holster/belt/update_icon()

Reply:
I don't mean the icon, but that `/obj/item/storage/pouch/holster/belt/update_icon()` adds overlays on top.
Then again, if that's fine on local...